scorearea.cpp

```diff
@@ -0,0 +1,51 @@
+#include "scorearea.h"
+
+#include "colliderbox.h"
+#include "pong.h"
+#include "rectangle.h"
+#include "scorecontroller.h"
+
+#include <spdlog/spdlog.h>
+
+namespace pong
+{
+
+ScoreArea::ScoreArea(float width, float height, bool playerSide)
+{
+    mMesh = std::make_unique<Rectangle>(width, height);
+    mColliderBox = std::make_unique<ColliderBox>(width, height);
+    mIsPlayerScoreArea = playerSide;
+    SetInstanceName("ScoreArea");
+}
+
+void ScoreArea::OnStart()
+{
+    mScoreController = Pong::FindGameObject<ScoreController>();
+    if (mScoreController == nullptr)
+    {
+        spdlog::error("ScoreController instance not found!");
+    }
+}
+
+void ScoreArea::OnCollisionStart(GameObject& other)
+{
+    if (other.GetInstanceName() == "Ball")
+    {
+        ASSERT(mScoreController != nullptr);
+
+        if (mIsPlayerScoreArea)
+        {
+            spdlog::info("Player scored!");
+            mScoreController->PlayerScored();
+        }
+        else
+        {
+            spdlog::info("Opponent scored!");
+            mScoreController->OpponentScored();
+        }
+
+        spdlog::info("New Score: P {} - O {}", mScoreController->GetPlayerScore(), mScoreController->GetOpponentScore());
```

Comment:
So I definitely want to get a real time logger in, but spdlog has no support for sequence numbers it seems. How would I be able to insert sequence numbers into these logs? I know we used macros for our logging, and I can imagine a way to do that by inserting an sequence number into the log call itself. Curious if you had anything off the top of your head.

Reply:
Yeah, a macro seems smart I think, just automatically extending the log line to put in a sequence number as it increments.

Alternatively look into making some constexpr function for it. Like macros, but type safe and may not bite you with some weirdness. They have some funny quirks that are good to learn.